libs/dbus_util/inc/public/tfc/dbus/sml_interface.hpp

```diff
@@ -0,0 +1,219 @@
+#pragma once
+
+#include <memory>
+#include <sstream>
+#include <string>
+#include <string_view>
+
+#include <boost/sml.hpp>
+
+#include <tfc/dbus/sdbusplus_fwd.hpp>
+#include <tfc/dbus/string_maker.hpp>
+#include <tfc/sml_logger.hpp>
+#include <tfc/stx/concepts.hpp>
+
+namespace tfc::dbus::sml {
+
+namespace tags {
+static constexpr std::string_view sub_path{ "StateMachines" };
+static constexpr std::string_view path{ const_dbus_path<sub_path> };
+}  // namespace tags
+
+namespace detail {
+
+struct interface_impl {
+  explicit interface_impl(std::shared_ptr<sdbusplus::asio::dbus_interface>);
+  interface_impl(interface_impl const&) = delete;
+  interface_impl(interface_impl&&) noexcept = default;
+  auto operator=(interface_impl const&) -> interface_impl& = delete;
+  auto operator=(interface_impl&&) noexcept -> interface_impl& = default;
+
+  void on_state_change(std::string_view source_state, std::string_view destination_state, std::string_view event);
+  void dot_format(std::string_view state_machine);
+
+  std::string source_state_{};
+  std::string destination_state_{};
+  std::string event_{};
+  std::string state_machine_dot_formatted_{};
+  std::shared_ptr<sdbusplus::asio::dbus_interface> dbus_interface_{};
+};
+
+template <class state_machine_t, class state_t>
+void dump(state_t const& current_state, std::ostream& out) noexcept;
+
+template <template <typename, typename> typename event_t, typename first_t, typename second_t>
+auto constexpr extract_event_type(event_t<first_t, second_t> const&) noexcept -> std::string {
+  return boost::sml::aux::get_type_name<second_t>();
+}
+
+template <typename event_t>
+auto constexpr extract_event_type(event_t const& event) noexcept -> std::string {
+  if constexpr (tfc::stx::is_specialization_v<event_t, boost::sml::back::on_entry> ||
+                tfc::stx::is_specialization_v<event_t, boost::sml::back::on_exit>) {
+    return detail::extract_event_type(event);
+  } else {
+    return boost::sml::aux::get_type_name<event_t>();
```

Comment:
I see, so you can override a name otherwise you get a sml provided name. That is pretty neat.

Reply:
Thank you, this is actually not doing the naming thing but it should! Will add.

Reply:
fixed here https://github.com/Skaginn3x/framework/pull/225/commits/51aa82e3e63a55f1ee621bc60cd81baeec6554ee